lib/core/buffer.cpp

```diff
@@ -52,14 +52,11 @@ GenericBuffer::GenericBuffer(const std::string& _buffer_name, const std::string&
 
 GenericBuffer::~GenericBuffer() {}
 
-bool GenericBuffer::set_metadata(int ID, metadataContainer* meta) {
+bool GenericBuffer::set_metadata(int ID, std::shared_ptr<metadataObject> meta) {
     assert(ID >= 0);
     assert(ID < num_frames);
     buffer_lock lock(mutex);
```

Comment:
What is this mutex used for? I thought it was only for reference counting. If so, then we wouldn't need it any more.